repository.go

```diff
@@ -81,6 +81,7 @@ func (r *Repository) Hook(f *Field) (err error) {
 			return fmt.Errorf("configuring field %s: missing key %s", f.Path, f.ConfigurationKey)
 		}
 		raw = def
+		provider = "default"
```

Comment:
We should probably provide this string as a const to avoid people mistyping it.

Reply:
Good point